### PR TITLE
core-vm: remove string allocations

### PR DESCRIPTION
**Pull request by tynes**
## Description

Removes string allocations in the hot code path

### Fixes
- Fixes # [Link to Issue]

## Contributing Agreement

- [x] I have read and understood the [Optimism Contributing Guide and Code of Conduct](./CONTRIBUTING.md) and am following those guidelines in this pull request.

### CODE REVIEW

File: core/vm/evm.go [hunk 1]
```diff
@@ -33,6 +32,14 @@ import (
 	"github.com/ethereum/go-ethereum/rollup/dump"
 )
 
+var deadPrefix, fourtyTwoPrefix, zeroPrefix []byte
```
Comment (smartcontracts) [inline core/vm/evm.go]:
```suggestion
// Will be removed when we update EM to return data in `run`.
var deadPrefix, fourtyTwoPrefix, zeroPrefix []byte
```

**Review by smartcontracts — APPROVED** [review]
LGTM

File: core/vm/evm.go [hunk 2]
```diff
@@ -33,6 +32,15 @@ import (
 	"github.com/ethereum/go-ethereum/rollup/dump"
 )
 
+// Will be removed when we update EM to return data in `run`.
+var deadPrefix, fourtyTwoPrefix, zeroPrefix []byte
```
Comment (tynes) [inline core/vm/evm.go]: its `forty` not `fourty`, will fix

**Comment by smartcontracts** [comment]
Appears to be working fine.

**Comment by smartcontracts** [comment]
Should be removed anyway when we get the execution manager to pass data up.

**Review by smartcontracts — APPROVED** [review]

**Pull request merged**